### ISSUE TITLE
Clarity about losses

### ISSUE THREAD

**Issue by s4sarath** [lucidrains/imagen-pytorch#45]
Hi Guys,

Thanks for the amazing work.
How many losses we have overall. 

1. MSE( BaseModel(x_noisy), x)
2. MSE( SuperResModel1(x_noisy), x) # 64 ----> 256
3. MSE( SuperResModel2(x_noisy), x) # 256 ----> 1024

So is it ```mean(L1 + L2 + L3)``` ? 

Or update based on ```each loss``` separately. I am little confused.

**Comment by lucidrains** [comment]
@s4sarath yup, you are correct, except each model is trained separately, so you wouldn't need to sum up all the losses like that

**Comment by legacyai** [comment]
Makes sense. So, one more last thing.
 When minimising L1, we update variables of Base Model (Unet0). 
Similarly, while minimising L2 ( Loss of second Unet1 (first resolution ) model ), we update variables of only Unet1 right. Base Model variables are freeze here right.

Because it has contributions from Base Model also right. So, thats why I am explicitly asking.

**Comment by lucidrains** [comment]
when training unet1, we actually don't condition on the output of unet0, we just take the image, resize it, and add some noise and pretend it is the output. so every unet in the cascading ddpm can be trained separately and then chained together at the end

**Comment by s4sarath** [comment]
Oh :-( . I thought this was all chained together conditioned on the output of previous one. 
So as per your comment, we can train a diffusion model ( unet0 ) conditioned on text separately. 

Then unet1 can be trained separately so on. Is this right. I thought in Cascading DDPM paper, they conditioned on the output of first diffusion model ( unet0 ). May be I was wrong 

Thanks a lot.

**Comment by lucidrains** [comment]
Yup, each stage can be trained separately, which is actually one of the main benefits of cascading ddpm!
